libfreerdp/core/gcc.c

```diff
@@ -2198,26 +2198,25 @@ BOOL gcc_write_client_monitor_data(wStream* s, const rdpMcs* mcs)
 			{
 				baseX = current->x;
 				baseY = current->y;
-				break;
 			}
 		}
 
 		for (UINT32 i = 0; i < settings->MonitorCount; i++)
 		{
 			const rdpMonitor* current = &settings->MonitorDefArray[i];
-			const UINT32 left = WINPR_ASSERTING_INT_CAST(uint32_t, current->x - baseX);
-			const UINT32 top = WINPR_ASSERTING_INT_CAST(uint32_t, current->y - baseY);
-			const UINT32 right = left + WINPR_ASSERTING_INT_CAST(uint32_t, current->width - 1);
-			const UINT32 bottom = top + WINPR_ASSERTING_INT_CAST(uint32_t, current->height - 1);
-			const UINT32 flags = current->is_primary ? MONITOR_PRIMARY : 0;
+			const INT32 left = WINPR_ASSERTING_INT_CAST(int32_t, current->x - baseX);
```

Comment:
`left = WINPR_ASSERTING_INT_CAST(` now useless, `current->x` already is of same type